### ISSUE TITLE
Add CI tests to check whether lbmap-related restore works

### ISSUE THREAD

**Issue by brb** [cilium/cilium#13867]
The https://github.com/cilium/cilium/issues/13824 issue revealed one weakness in the `k8sT/Services.go` suite - we don't test restore procedures of the lbmap-related maps. The unit tests in `pkg/service/service_test.go` contain test cases for testing the restore procedures, but they are relying on the lbmap mock implementation, so they cannot catch such issues as #11213 

~Therefore, we should extend the CI suite by adding a case in which cilium-agent gets restarted and is able to function.~ We should add a _privileged_ unit test which uses the actual lbmap implementation.

**Comment by pchaigno** [comment]
@brb This is just a matter of taking one of the existing BPF service handling tests (e.g., [`Tests NodePort`](https://github.com/cilium/cilium/blob/c69dfd106131f174ec315b8522b36405b67207f2/test/k8sT/Services.go#L2138)) and check that it still work during and after Cilium restart, similarly to [what is done for FQDN policies](https://github.com/cilium/cilium/blob/c69dfd106131f174ec315b8522b36405b67207f2/test/k8sT/fqdn.go#L115), right?

**Comment by brb** [comment]
Yep. Ideally, the test should involve a majority of the lbmap-related maps (svc, affinity, src-range).

**Comment by oneiro-naut** [comment]
Hi, I would like to work on this issue. Can this be assigned to me?

**Comment by pchaigno** [comment]
@oneiro-naut I assigned you. Don't hesitate to reach out on Slack (e.g., #development or #testing) if you have any question.

**Comment by oneiro-naut** [comment]
Hi @pchaigno. Since the cilium daemon itself calls RestoreServices()[1] so we don't explicitly need to call the service restore function in the test cases. Is that correct?

[1]https://github.com/cilium/cilium/blob/fb85fee7001e7fcc9fc3466c4ff018fe787fe26c/daemon/cmd/daemon.go#L498

**Comment by joestringer** [comment]
@brb I wonder is this still relevant following the LB rework in v1.18?

**Comment by brb** [comment]
The new implementation no longer restores all fields except for the SVC IDs. According to @joamaki the following tests check for the ID restoration:

- https://github.com/cilium/cilium/blob/main/pkg/loadbalancer/tests/testdata/quarantined.txtar
- https://github.com/cilium/cilium/blob/main/pkg/loadbalancer/tests/testdata/migrate-any-proto.txtar

Let's close the issue.
